core/src/test/java/io/confluent/kafka/schemaregistry/rest/RestApiRegisterSchemaTagsTest.java

```diff
@@ -222,4 +251,54 @@ public void testRegisterSchemaTagsWithInvalidTags() throws Exception {
       assertEquals(Errors.INVALID_SCHEMA_ERROR_CODE, e.getErrorCode());
     }
   }
+
+  @Test
+  public void testRegisterSchemaTagsIncrementalRuleSet() throws Exception {
+    String subject = "test";
+    TestUtils.registerAndVerifySchema(restApp.restClient, schemaString, 1, subject);
+
+    TagSchemaRequest tagSchemaRequest = new TagSchemaRequest();
+    tagSchemaRequest.setNewVersion(2);
+    Rule rule = new Rule("myRule", null, null, null,
+        "fooType", ImmutableSortedSet.of("PII"), null, null, null, "NONE,NONE", false);
+    Rule rule2 = new Rule("myRule2", null, null, null,
+        "fooType", ImmutableSortedSet.of("PII"), null, null, null, "NONE,NONE", false);
+    Rule rule3 = new Rule("myRule3", null, null, null,
+        "fooType", ImmutableSortedSet.of("PII"), null, null, null, "NONE,NONE", false);
+    RuleSet ruleSet = new RuleSet(Collections.emptyList(), ImmutableList.of(rule, rule2, rule3));
+    tagSchemaRequest.setRulesToMerge(ruleSet);
+    tagSchemaRequest.setRulesToRemove(ImmutableList.of("myRule4"));
+
+    RegisterSchemaResponse responses = restApp.restClient
+        .modifySchemaTags(RestService.DEFAULT_REQUEST_PROPERTIES, tagSchemaRequest, subject, "latest");
+    assertEquals(2, responses.getId());
+    assertEquals(ruleSet, responses.getRuleSet());
+
+    Schema result = restApp.restClient.getLatestVersion(subject);
+    assertEquals("2", result.getMetadata().getProperties().get("confluent:version"));
+    assertEquals(ruleSet, responses.getRuleSet());
+
+    tagSchemaRequest = new TagSchemaRequest();
+    tagSchemaRequest.setNewVersion(3);
+    Rule rule5 = new Rule("myRule5", null, null, null,
+        "fooType", ImmutableSortedSet.of("PII2"), null, null, null, "NONE,NONE", false);
+    Rule rule4 = new Rule("myRule4", null, null, null,
+        "fooType", ImmutableSortedSet.of("PII2"), null, null, null, "NONE,NONE", false);
+    rule2 = new Rule("myRule2", null, null, null,
+        "fooType", ImmutableSortedSet.of("PII2"), null, null, null, "NONE,NONE", false);
+    ruleSet = new RuleSet(Collections.emptyList(), ImmutableList.of(rule5, rule4, rule2));
```

Comment:
All tests assume empty migration rules. Should we add a test case for removal from migrationRules?

Reply:
Sure, I'll add a teset